.github/workflows/build.yml

```diff
@@ -6,21 +6,205 @@ on:
 
+defaults:
+  run:
+    shell: bash
+
 jobs:
   build:
-
     runs-on: ubuntu-latest
+    container: collectd/ci:${{ matrix.container_tag }}
+    continue-on-error: ${{ matrix.allow_failures }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # for tasks that are optional, use the continue-on-error option, to prevent a workflow from failing when the task fails
+        allow_failures: [ false ]
+        container_tag:
+          # debian family
+          - buster_amd64
+          - stretch_amd64
+          - stretch_i386
+          - trusty_amd64
+          - xenial_amd64
+          # RedHat family
+          - el7_x86_64
+          - fedora28_x86_64
+        config_flags: ['']
+    env:
+      MAKEFLAGS: "-j 2"
+      CONFIGURE_FLAGS: ${{ matrix.config_flags }}
+    steps:
+    - uses: actions/checkout@v2
+    - run: type pkg-config
+    - run: pkg-config --list-all | sort -u
+    - name: Generate configure script
+      run:
+        ./build.sh
+    - name: configure
+      run: ./configure $CONFIGURE_FLAGS
+    - name: Make
+      run: make
+    - name: make check
+      run: make check
+    - name: install bzip2
+      continue-on-error: true
+      run: |
+        yum install -y bzip2 || apt install -y bzip2
+    - name: make
+      continue-on-error: true
+      run: |
+        make distcheck DISTCHECK_CONFIGURE_FLAGS="--disable-dependency-tracking --enable-debug"
 
+  experimental-debian:
+    runs-on: ubuntu-20.04
+    container: collectd/ci:${{ matrix.container_tag }}
+    continue-on-error: true
+    strategy:
+      fail-fast: false
+      matrix:
+        # for tasks that are optional, use the continue-on-error option, to prevent a workflow from failing when the task fails
+        container_tag:
+          - sid_amd64
+        config_flags: ["--disable-dpdkstat --disable-dpdkevents --disable-virt"]
+    env:
+      MAKEFLAGS: "-j 2"
+      CONFIGURE_FLAGS: ${{ matrix.config_flags }}
     steps:
     - uses: actions/checkout@v2
+    - run: type pkg-config
+    - run: pkg-config --list-all | sort -u
     - name: Generate configure script
       run:
         ./build.sh
     - name: configure
-      run: ./configure
+      run: ./configure $CONFIGURE_FLAGS
+    - name: Make
+      run: make
+    - name: make check
+      run: make check
+    - name: install bzip2
+      continue-on-error: true
+      run: |
+        yum install -y bzip2 || apt install -y bzip2
+    - name: make
+      continue-on-error: true
+      run: |
+        make distcheck DISTCHECK_CONFIGURE_FLAGS="--disable-dependency-tracking --enable-debug"
+
+  experimental-fedora:
+    runs-on: ubuntu-20.04
+    container: ${{ matrix.container_image }}
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # for tasks that are optional, use the continue-on-error option, to prevent a workflow from failing when the task fails
+        container_image:
+          - fedora:34
+          - fedora:rawhide
```

Comment:
```suggestion
```

Reply:
I'll add this in.

There are a few changes I'm going to make, since I don't have to add in the dependency install steps anymore:

* Move fedora34 into the ``build`` job
* Combine the ``experimental-debian`` and ``experimental-fedora`` jobs into an ``experimental`` job
* Fedora rawhide will be in the ``experimental`` job, once it works again.
* Add in the new containers from (collectd/ci-docker) to the build jobs